src/components/Deposit.js

```diff
@@ -1,117 +1,137 @@
-            setTimeout(() => setStatus(''),3000);
-            return false;
-        }
-        return true;
-    }
+  const [deposit, setDeposit] = React.useState();
+  const [show, setShow] = React.useState(true);
+  const [status, setStatus] = React.useState('');
+  const ctx = React.useContext(UserContext);
+  const users = ctx.state.users;
+  //balance fetches from context
+
+  if (!users.length) {
+    console.error('No users!');
+  }
+
+  let balance = users[users.length - 1].balance;
 
-    function validateNumber(field) {
-        if (isNaN(parseFloat(field))) {
-            setStatus('Error');
-            setTimeout(() => setStatus(''),3000);
-            return false;
-        }
-        return true;
+  function validate(field, label) {
+    if (!field) {
+      setStatus('Error: ' + label);
+      setTimeout(() => setStatus(''), 3000);
+      return false;
     }
+    return true;
+  }
 
-    function validatePositiveNumber(field) {
-        if (Number(field) < 0) {
-            setStatus('Error');
-            setTimeout(() => setStatus(''),3000);
-            return false;
-        }
-        return true;
+  function validateNumber(field) {
+    if (isNaN(parseFloat(field))) {
+      setStatus('Error');
+      setTimeout(() => setStatus(''), 3000);
+      return false;
     }
+    return true;
+  }
 
-    function handleDeposit() {
-        if(!validate(deposit, 'deposit')) {
-            alert('Deposit amount is required');
-            return;
-        }
+  function validatePositiveNumber(field) {
+    if (Number(field) < 0) {
+      setStatus('Error');
+      setTimeout(() => setStatus(''), 3000);
+      return false;
+    }
+    return true;
+  }
 
-        if(!validateNumber(deposit)) {
-            alert('Deposit amount must be a number');
-            return;
-        }
+  function handleDeposit() {
+    if (!validate(deposit, 'deposit')) {
+      alert('Deposit amount is required');
+      return;
+    }
 
-        if(!validatePositiveNumber(deposit)) {
-            alert('Deposit amount cannot be negative');
-            return;
-        }
-   
-        let newBalance = balance + Number(deposit);
-        
-        console.log('new balance is ' + newBalance);
-        setShow(false);
-        return newBalance;
-        }
+    if (!validateNumber(deposit)) {
+      alert('Deposit amount must be a number');
+      return;
+    }
 
-    function clearForm(){
-        setDeposit('');
-        setShow(true);
+    if (!validatePositiveNumber(deposit)) {
+      alert('Deposit amount cannot be negative');
+      return;
     }
 
+    let newBalance = balance + Number(deposit);
+    if (ctx.state.users.length) {
+      // Find the last user added and set their balance to the new balance
+      ctx.setUsers(
+        ctx.state.users.map((user, index) =>
+          index === ctx.state.users.length - 1
+            ? { ...user, balance: newBalance }
+            : user
+        )
+      );
+    }
```

Comment:
Here we first check to make sure the length of users isn't zero. If it is then there are no users for us to update. Then we tell context to update user state with `ctx.setUsers`.
We loop over all of the existing users and determine which user needs to be modified. In this case we're looking for the most recently created user, hence `index === users.length - 1`. If the user isn't the most recent then we don't need to make any changes to it and can just leave that as it is (`: user`). But if it is the last user then we need to update the balance to `newBalance`, hence `? {...user, balance: newBalance}`.


I made a mistake here, this should be:
```suggestion
    if (users.length) {
      // Find the last user added and set their balance to the new balance
      ctx.setUsers(
       users.map((user, index) =>
          index === users.length - 1
            ? { ...user, balance: newBalance }
            : user
        )
      );
    }
```